src/AdminTab/TableList.tsx

```diff
@@ -50,9 +51,41 @@ const TableDropdown: FunctionComponent<TableDropdownProps> = ({ table, onOpenMet
       <Menu.Item
         icon={<BarChartOutlined />}
         key={`${table.name}-examples`}
+        disabled={examplesInProgress}
         onClick={async () => {
+          const messageKey = getUniqueKey();
+
+          const messageTimeout = setTimeout(
+            () =>
+              message.loading({
+                content: t('Examples for {{tableName}} under construction...', { tableName: table.name }),
+                key: messageKey,
+                duration: 0,
+              }),
+            1000,
+          );
+
+          setExamplesInProgress(true);
+          const startTime = performance.now();
           const examplesCollectionId = await getTableExamples(table).result;
-          onOpenMetabaseTab(`collection/${examplesCollectionId}`);
+          const elapsed = performance.now() - startTime;
```

Comment:
I think `process.hrtime` is the more "classic" way to measure time in node.

Reply:
we're using it elsewhere (`runTask` function). Should we have two ways or change over there as well?